### PR TITLE
Gcp HTTP functions improvements

### PR DESCRIPTION
**Pull request by loicmathieu**
Small enhancement to the Google Cloud Functions HTTP bindings:
- Add a test (on the RESTEasy layer, not using the Function invoker)
- Set the extension as exeprimental (as the other ones)
- Packaging improvements

### CODE REVIEW

**Comment by loicmathieu** [comment]
@patriot1burke @gsmet I forgot to send this PR, sorry, it's a bit later but it align the GCF HTTP binging with the other extension (regarding deployment) and more important, it set this extension as experimental so I would like it to be merged asap and backported to 1.6.

File: docs/src/main/asciidoc/gcp-functions-http.adoc [hunk 1]
```diff
@@ -140,33 +142,18 @@ public class GreetingRoutes {
...
-To deploy your JAR to Google Cloud, you need to pass a directory with only this JAR inside it, to the `gcloud` utility.
-
-So first, create a `deployment` directory and copy the generated artifact inside it.
-
-[source]
-----
-mkdir deployment
-cp target/google-cloud-functions-http-1.0-SNAPSHOT-runner.jar deployment/
-----
+The result of the previous command is a single JAR file inside the `target/deployment` repository that contains classes and dependencies of the project.
```
Comment (gsmet) [inline docs/src/main/asciidoc/gcp-functions-http.adoc:148]:
```suggestion
The result of the previous command is a single JAR file inside the `target/deployment` directory that contains the classes and the dependencies of the project.
```

File: docs/src/main/asciidoc/gcp-functions-http.adoc [hunk 2]
```diff
@@ -140,33 +142,18 @@ public class GreetingRoutes {
...
 
 Then you will be able to use `gcloud` to deploy your function to Google Cloud.
 
 [source]
 ----
 gcloud beta functions deploy quarkus-example-http \
   --entry-point=io.quarkus.gcp.functions.http.QuarkusHttpFunction \
-  --runtime=java11 --trigger-http --source=deployment
+  --runtime=java11 --trigger-http --source=target\deployment
```
Comment (gsmet) [inline docs/src/main/asciidoc/gcp-functions-http.adoc:156]: just to be sure, it's normal it's a `\` and not a `/`?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few minor suggestions. Looks safe to backport.

File: docs/src/main/asciidoc/gcp-functions-http.adoc (hunk 2 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/gcp-functions-http.adoc:156]: No, I'll change it

Comment (loicmathieu) [reply docs/src/main/asciidoc/gcp-functions-http.adoc]: done

**Comment by loicmathieu** [comment]
@gsmet PR updated based on your requested changes.

**Review by gsmet — APPROVED** [review]

**Pull request merged**